### ISSUE TITLE
Need a way to ignore ORA errors and continue executing other stmts in a file

### ISSUE THREAD

**Issue by ThiyaguE** [flyway/flyway#1175]
Very often developers , would fix simple uniq constraint violation kind of errors , offline / after deployment. Having 1000s of insert/update stmts in set of files, if flyway stops on encountering error and refuse to execute remaining stmts in file, we feel it creates delay in prod deployment ( with limited window time). Need a flag ( like -X for extended debug ) , when we set, flyway could continue executing other stmts in a file.  it will be better , if we have a way to specify what kind of error codes can be ignored. like a separate file ( like callback) for ignorable error codes.

As of now, we have work around to completely skip one file ( by changing its prefix/suffix). THere is no way to skip erring stmts in a file.

This will help dev/ops teams to complete prod deployment with in scheduled time, and CI/CD will be easy.

**Comment by axelfontaine** [comment]
This is a niche requirement and not something we plan to add to the main distribution. You can however achieve this by writing your own MigrationResolver/MigrationExecutor.

**Comment by bladeoflight16** [comment]
This is less a "niche requirement" as much as it is just a bad idea. Consider:

  * If flyway reports success even though some statements errored out, you *don't know the state of your database* once Flyway completes. Silent failures are more likely to lead to errors in prod following the deployment because applications or users assume the database is in a particular state that it isn't. Delays in prod release are far preferable to *failures in prod* after release.
  * If you're getting `INSERT` conflicts, you probably have a bug in your migrations. You're probably just not tracking what you've already inserted, but if you really need to conditionally insert, you can do so with filters or by combining an `INSERT` and an `UPDATE` (or using an upsert type statement if your database supports it).
  * You can detect these failures ahead of time by testing against a fresh, scrubbed (private data removed) clone of the database from production.
  * In most databases, you could already work around this using the procedural language the database provides. I know first hand that blocks of pl/pgsql in PostgreSQL and PL/SQL in Oracle have a "catch" mechanism. If you *really* need to pass over errors, you can use these. T-SQL in SQL Server has a try/catch mechanism as well. You could also just check a condition before executing the command in the with the `IF` blocks these languages provide. I'd be surprised if there's a modern database that doesn't support something along these lines.

Bottom line: you don't need Flyway to catch your errors. If probably a bad idea to ignore them, and it's possible to do with just queries when it's not a bad idea.
